packages/experimental-package/src/components/RemoveModal/RemovalModal.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useRef } from 'react';
+import { Modal, TextInput } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import { expPrefix } from '../../global/js/settings';
+import uuidv4 from '../../global/js/utils/uuidv4';
+
+export const RemovalModal = ({
+  body,
+  className,
+  header,
+  iconDescription,
+  id,
+  inputInvalidText,
+  inputLabelText,
+  inputPlaceholderText,
+  onRequestClose,
+  onRequestSubmit,
+  open,
+  primaryButtonText,
+  resource,
+  secondaryButtonText,
+  subheader,
+  textConfirmation,
+}) => {
+  const [userInput, setUserInput] = useState('');
+  const idRef = useRef(id || uuidv4());
+  const onChangeHandler = (e) => {
+    setUserInput(e.target.value);
+  };
+  const primaryButtonDisabled = textConfirmation && userInput !== resource;
+  return (
+    <Modal
+      danger
+      className={`${expPrefix}-removal-modal ${className}`}
+      iconDescription={iconDescription}
+      open={open}
+      modalHeading={header}
+      modalLabel={subheader}
+      primaryButtonText={primaryButtonText}
+      primaryButtonDisabled={primaryButtonDisabled}
+      secondaryButtonText={secondaryButtonText}
+      onRequestSubmit={onRequestSubmit}
+      onRequestClose={onRequestClose}>
+      <p className={`${expPrefix}-removal-modal-body`}>{body}</p>
+      {textConfirmation && (
+        <TextInput
+          id={idRef.current}
```

Comment:
This alongside other suggestion. If the user passes an Id we must apply to the outer element as per standard HTML.

```suggestion
          id={`${idRef.current}-confirm-input`}
```